lib/fluent/supervisor.rb

```diff
@@ -268,7 +268,8 @@ def initialize(path, level, chuser, chgroup, opts)
 
       def init
         if @path && @path != "-"
-          @io = File.open(@path, "a")
+          Dir.mkdir(File.dirname(@path)) unless File.exists?(@path)
```

Comment:
It fails when the file (`@path`) doesn't exist and directory (`File.dirname(@path)`) already exists, and when the parent directories of directory (`File.dirname(File.dirname(@path))`) doesn't exist.
It's better to use `FileUtils.mkdir_p(File.dirname(@path)) unless Dir.exist?(File.dirname(@path))`